source/Symbol/SwiftASTContext.cpp

```diff
@@ -1591,7 +1591,10 @@ lldb::TypeSystemSP SwiftASTContext::CreateInstance(lldb::LanguageType language,
                      osi < ose; ++osi) {
                   const char *clang_argument =
                       ast_context->GetClangArgumentAtIndex(osi);
-                  swift_ast_sp->AddClangArgument(clang_argument, true);
+                  bool force = true;
+                  if (clang_argument && clang_argument[0] == '-' && clang_argument[1] == 'D')
+                    force = false;
+                  swift_ast_sp->AddClangArgument(clang_argument, force);
```

Comment:
Don't forget that people write macros as both `-DFOO` and `-D FOO`. And don't forget about `-U` as well! (But I think it's okay to not worry about -D and -U fighting.)